src/java/org/apache/cassandra/concurrent/FutureTask.java

```diff
@@ -89,6 +89,34 @@ protected boolean tryFailure(Throwable t)
 
     public static <T> Callable<T> callable(Runnable run)
     {
+        if (run instanceof DebuggableTask.RunnableDebuggableTask)
+        {
+            return new DebuggableTask.CallableDebuggableTask<T>()
+            {
+                final RunnableDebuggableTask task = (RunnableDebuggableTask) run;
+                public T call()
+                {
+                    run.run();
+                    return null;
+                }
+
+                public long creationTimeNanos()
+                {
+                    return task.creationTimeNanos();
+                }
+
+                public long startTimeNanos()
+                {
+                    return task.startTimeNanos();
+                }
+
+                public String description()
+                {
+                    return task.description();
+                }
+            };
+        }
```

Comment:
Aside: I think there's actually a chance the changes I made to `TaskFactory` in CASSANDRA-19577 overlap w/ this. In other words, I think `callable()` doing this might actually make my `TaskFactory` changes unnecessary. I don't think our changes will break each other though...

Reply:
I think I have first tried to do it very similarly to how you did it, but this has ported better to other branches. Please let me know how you want to proceed.

Reply:
As long as the tests I wrote in 19577 still pass, I'm 100% fine if you want to revert those bits from my patch, unless doing it that way entails more object creation.